mapflow/functional/layer_utils.py

```diff
@@ -370,6 +384,33 @@ def download_results_file_callback(self, response: QNetworkReply, path: str) ->
             f.write(response.readAll().data())
         self.message_bar.pushSuccess(self.tr("Results saved"),
                                      self.tr("see in {path}!").format(path=path))
+        
+    def download_aoi_file_callback(self, response: QNetworkReply, path: str) -> None:
+        """
+        Write area of interest to the geojson file
+        """
+        self.dlg.saveResultsButton.setEnabled(True)
+        with open(path, mode='wb') as f:
```

Comment:
We don't need to write-read-write again.

Just parse the response data as JSON, then extract AOIs geometries and features, then dump() it to the file.